### PR TITLE
FR-15371 - Example app makeover

### PR DESCRIPTION
**Pull request by doregg**
Most of the files just copied form the admin box example app

Main changes compared to the admin box:
- Full stepped up page replaced by the HOC page
- React is 17, the main render method is like before

Screenshot 2024-02-29 at 14 29 02

### CODE REVIEW

File: packages/demo-saas/src/entitlements/components/EntitlementsBase.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  isEntitled: Entitlement['isEntitled'];
+  justification?: Entitlement['justification'];
+}) => {
+  return (
+    <Box
+      display='flex'
+      flexDirection='column'
+      className={`entitlement-item ${isEntitled ? 'entitlement-item-entitled' : 'entitlement-item-not-entitled'}`}
+    >
+      <Box
```
[the remark this answers is not in the document]
Comment (doregg) [reply packages/demo-saas/src/entitlements/components/EntitlementsBase.tsx:20]: #cycode_sast_ignore_here

File: packages/demo-saas/src/stepUp/Wallet/utils.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+export const getQueryParam = (key: string) => new URLSearchParams(window.location.search).get(key);
+
+export const addLocalStorage = (key: string, value: string) => localStorage.setItem(key, value);
+export const getLocalStorage = (key: string) => localStorage.getItem(key);
+export const removeLocalStorage = (...keys: string[]) => keys.forEach((key) => localStorage.removeItem(key));
+
+export function getRandomInt(min: number, max: number): number {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
```
[the remark this answers is not in the document]
Comment (doregg) [reply packages/demo-saas/src/stepUp/Wallet/utils.ts:39]: #cycode_sast_ignore_here

File: packages/demo-saas/src/BaseHomePage/components/CheckoutDialogButton.tsx [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
```
Comment (yuvalotem1) [inline packages/demo-saas/src/BaseHomePage/components/CheckoutDialogButton.tsx:1]: maybe remove?

File: packages/demo-saas/src/BaseHomePage/components/AdminBoxButton.tsx [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { AdminPortal } from '@frontegg/js';
+import { DemoButton } from '../../DemoButton';
+
+export const AdminBoxButton = () => (
```
Comment (yuvalotem1) [inline packages/demo-saas/src/BaseHomePage/components/AdminBoxButton.tsx:5]: adminPortal

File: packages/demo-saas/src/BaseHomePage/components/Links.tsx [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  { route: '/', label: 'Home page' },
+  { route: '/entitlements', label: 'Entitlements' },
+  { route: '/step-up-high-max-age', label: 'Step up high max age' },
+  { route: '/step-up-small-max-age', label: 'Step up small max age' },
+  { route: '/step-up-no-max-age', label: 'Step up no nax age' },
+  { route: '/step-up-modals', label: 'Step up modals' },
+  { route: '/step-up-hoc', label: 'Step up HOC' },
+  { route: '/step-up-transfer', label: 'Step up transfer' },
+  { route: '/unknown-route', label: 'Fallback route' },
+  { route: '/test', label: 'Old test' },
```
Comment (yuvalotem1) [inline packages/demo-saas/src/BaseHomePage/components/Links.tsx:27]: you can reuse the strings from paths in routes - https://github.com/frontegg/frontegg-react/pull/1118/files#diff-4d1c0f1c53695db58cc90854af15a62e896842473b052336143c2a9f04788299R51-R58

File: packages/demo-saas/src/FronteggProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
+import React, { FC, ReactNode, useCallback, useMemo } from 'react';
```
Comment (yuvalotem1) [inline packages/demo-saas/src/FronteggProvider.tsx:1]: remove, we already have it in the react SDK

File: packages/demo-saas/src/App.tsx [hunk 7]
```diff
@@ -1,15 +1,86 @@
-import React, { FC } from 'react';
-import { AppHeader } from './components/AppHeader';
-import { AppSidebar } from './components/AppSidebar';
-import { AppContent } from './components/AppContent';
+import React, { FC, useState } from 'react';
+import { BrowserRouter, Route, Switch } from 'react-router-dom';
+import Loader from '@mui/material/CircularProgress';
+import Box from '@mui/material/Box';
+import { FronteggProvider } from './FronteggProvider';
```
Comment (yuvalotem1) [inline packages/demo-saas/src/App.tsx:5]:
```suggestion
import { FronteggProvider } from '@frontegg/react';
```

File: packages/demo-saas/src/App.tsx [hunk 8]
```diff
@@ -1,15 +1,86 @@
...
+import { FronteggAppOptions } from '@frontegg/types';
+import { authOptions } from './customizationOptions/authOptions';
```
Comment (yuvalotem1) [inline packages/demo-saas/src/App.tsx:7]: take also from auth types

File: packages/demo-saas/src/FronteggProvider.tsx
Comment (yuvalotem1) [inline packages/demo-saas/src/FronteggProvider.tsx:1]: remove, we already have it in the react SDK

File: packages/demo-saas/src/adminBoxPreview/AdminBoxIframe.tsx
Comment (yuvalotem1) [inline packages/demo-saas/src/adminBoxPreview/AdminBoxIframe.tsx:1]: also remove

File: packages/demo-saas/src/adminBoxPreview/AdminBoxInstance.tsx
Comment (yuvalotem1) [inline packages/demo-saas/src/adminBoxPreview/AdminBoxInstance.tsx:1]: also remove

File: packages/demo-saas/src/customizationOptions/themeOptions/splitMode.ts
Comment (yuvalotem1) [inline packages/demo-saas/src/customizationOptions/themeOptions/splitMode.ts:1]: Consider which customizations we want to hold here

File: packages/demo-saas/src/loginBoxPreview/LoginBoxIframe.tsx
Comment (yuvalotem1) [inline packages/demo-saas/src/loginBoxPreview/LoginBoxIframe.tsx:1]: remove this as well

File: packages/demo-saas/src/loginBoxPreview/LoginBoxPreview.tsx
Comment (yuvalotem1) [inline packages/demo-saas/src/loginBoxPreview/LoginBoxPreview.tsx:1]: remove this as well

File: packages/demo-saas/src/loginBoxPreview/LoginBoxPreviewContext.tsx
Comment (yuvalotem1) [inline packages/demo-saas/src/loginBoxPreview/LoginBoxPreviewContext.tsx:1]: remove this as well

File: packages/demo-saas/src/loginBoxPreview/index.ts
Comment (yuvalotem1) [inline packages/demo-saas/src/loginBoxPreview/index.ts:1]: remove this as well

File: packages/demo-saas/src/stepUp/Wallet/MyBalance.tsx [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  useEffect(() => {
+    setTimeout(() => {
+      setBalance(getRandomInt(1000, 2000));
+      setIsLoaded(true);
+    });
+  }, [setBalance, setIsLoaded]);
+
+  return isLoaded ? (
+    <div>
+      Your Amazon balance is <b style={{ color: '#1665c0' }}>{balance}</b>!
```
Comment (yuvalotem1) [inline packages/demo-saas/src/stepUp/Wallet/MyBalance.tsx:17]:
```suggestion
      Your Frontegg balance is <b style={{ color: '#1665c0' }}>{balance}</b>!
```

File: packages/demo-saas/src/stepUp/Wallet/QPWallet.tsx
Comment (yuvalotem1) [inline packages/demo-saas/src/stepUp/Wallet/QPWallet.tsx:1]: I would consider using the wallet terminology as it out of frontegg context

File: packages/demo-saas/src/App.tsx [hunk 10]
```diff
@@ -1,15 +1,84 @@
...
+import { ROUTE_PATHS } from './BaseHomePage/components/Links';
+
+const IS_HOSTED_LOGIN = true;
+
+const fronteggOptions: FronteggAppOptions =
+  // @ts-ignore
+  window.CYPRESS_CONFIG ||
+  ({
+    contextOptions: {
+      // baseUrl: process.env.PUBLIC_URL || process.env.REACT_APP_BASE_URL || DEFAULT_BASE_URL,
```
Comment (yuvalotem1) [inline packages/demo-saas/src/App.tsx:28]:
```suggestion
       baseUrl: process.env.PUBLIC_URL || process.env.REACT_APP_BASE_URL || DEFAULT_BASE_URL,
```

File: packages/demo-saas/src/App.tsx [hunk 11]
```diff
@@ -1,15 +1,84 @@
...
+      // clientId: process.env.REACT_APP_CLIENT_ID,
+    },
+
+    // guidesCdnUrl: 'https://assets.frontegg.com/admin-box/embedded-guides/1.0.0',
```
Comment (yuvalotem1) [inline packages/demo-saas/src/App.tsx:32]:
```suggestion
```

File: packages/demo-saas/src/App.tsx [hunk 12]
```diff
@@ -1,15 +1,84 @@
...
+          <Route path={ROUTE_PATHS.STEP_UP_TRANSFER} exact render={TransferStepUpPage} />
+          {/* For tests that use someurl as the authenticated-url */}
+          <Route path={'/someurl'} exact render={NotAFronteggPage} />
+          <Route
+            path={'/test'}
+            exact
+            component={() => {
+              return <div>Test</div>;
+            }}
+          />
```
Comment (yuvalotem1) [inline packages/demo-saas/src/App.tsx:65]: why do we need this route?

File: packages/demo-saas/src/BaseHomePage/components/NoBaseUrlSection/NoBaseUrlSection.css
Comment (yuvalotem1) [inline packages/demo-saas/src/BaseHomePage/components/NoBaseUrlSection/NoBaseUrlSection.css:1]: If you installed MUI why not using it? https://github.com/frontegg/frontegg-react/pull/1118/files#diff-c29d0d8bbd7609744f1df6cb234b298498394cb03d4a01d008eca2bb6c2c0792R17-R20

File: packages/demo-saas/src/entitlements/components/EntitlementBoxes.css
Comment (yuvalotem1) [inline packages/demo-saas/src/entitlements/components/EntitlementBoxes.css:1]: you can also use mui here

File: packages/demo-saas/src/index.tsx [hunk 13]
```diff
@@ -1,18 +1,10 @@
...
 ReactDOM.render(
-  <BrowserRouter>
-    <FronteggProvider
-      contextOptions={{
-        baseUrl: `https://david.frontegg.com`,
-      }}
-    >
-      <App />
-    </FronteggProvider>
-  </BrowserRouter>,
```
Comment (yuvalotem1) [inline packages/demo-saas/src/index.tsx:16]: maybe we should keep `BrowserRouter` for testing?

**Review by yuvalotem1 — APPROVED** [review]

File: packages/demo-saas/src/App.tsx (hunk 12 above)
Comment (doregg) [reply packages/demo-saas/src/App.tsx:65]: we want to test fallback route. We had a lot of issues regarding it.. and it reflects a real customer app

File: packages/demo-saas/src/BaseHomePage/components/NoBaseUrlSection/NoBaseUrlSection.css
Comment (doregg) [reply packages/demo-saas/src/BaseHomePage/components/NoBaseUrlSection/NoBaseUrlSection.css:1]: didn't want to invest time on it. It was the base code

File: packages/demo-saas/src/index.tsx (hunk 13 above)
Comment (doregg) [reply packages/demo-saas/src/index.tsx:16]: yup we have it in App.tsx

**Pull request merged**